Upgrade to electron 9.3.5

sadick254 opened the pull request:
This upgrades atom from `6.1.12` to `9.3.5`

## TODO
## Issues to resolve
 - [x] Failure due to node-abi not detecting abi for version 9.3.5 and runtime electron

## Failing tests

## Regressions to fix before merging


## Known bugs resolved by this upgrade



## Electron Upgrade Test Manifest

- [x] Review issues tagged with [`electron`](https://github.com/search?utf8=%E2%9C%93&q=org%3Aatom+label%3Aelectron+is%3Aopen+is%3Aissue) to see if any of them are fixed
- [x] Verify that you can collaborate with Teletype
    - [x] Host using Atom's current stable release can collaborate with guest using Atom with the new version of Electron
    - [x] Guest using Atom's current stable release can collaborate with host using Atom with the new version of Electron
    - [x] Host and guest can collaborate when both are using Atom with the new version of Electron
    <details>
    <summary>Steps</summary>

    1. Open a file
    1. Share a portal
    1. Have a guest join the portal
    1. When guest edits the text in the file, verify that the host sees those changes
    </details>
## Check for regressions experienced in previous upgrades

### Text Input/Keybindings

See how to [setup keyboard layouts](https://github.com/atom/maintainers-log/blob/master/how-tos/setup-keyboard-layouts.md).

- [x] IME not working (https://github.com/atom/atom/issues/14911)
	- [x] macOS
	- [x] Windows 10
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Install an IME layout, for example Pinyin simplified  
  1. Open Atom and use the layout you installed as the keyboard layout
  1. Example, if you picked Pinyin simplified type in `zhongwenshuru`

  **Expected:** in Pinyin simplified `中文输入` is expected

  ![gif](https://user-images.githubusercontent.com/14314532/27643849-ccf636da-5c54-11e7-9111-e7b09469cf10.gif)

  **Issues we've seen:** Only shows latin first character and not every character typed.

  ![gif](https://user-images.githubusercontent.com/14314532/27643959-0f04396e-5c55-11e7-838c-ac15ef580fbe.gif)
  </details>

- [x] macOS: Key binding with composition characters are ignored (https://github.com/atom/atom/issues/15189)
  <details>
  <summary>Repro steps</summary>

  1. Open your keymap.cson file and add the following:
      ```cson
      'atom-text-editor':
        # use hjkl; vim-like when pressing alt (word by word, paragraph by paragraph)
        # or alt (letter by letter)
        'alt-h': 'core:move-left'
        'alt-j': 'core:move-down'
        'alt-k': 'core:move-up'
        'alt-l': 'core:move-right'
      ```
  1. Save keymap.cson and reload atom.
  1. Try the new keymappings under `ABC - Extended` keyboard layout

  **Expected**: the keybindings work

  **Issues we've seen:** it types out the modified keys and ignores the key mapping
  </details>

- [x] macOS: Composition characters mess up insertion point (https://github.com/atom/atom/issues/15344)
  <details>
  <summary>Repro steps</summary>

  1. Open keymap.cson and add the following:
      ```cson
      'atom-text-editor':
        'alt-n': 'core:move-down'
        'alt-o': 'core:move-down'
      ```
  1. Have the following in your buffer:
      ```
      Foo
      Bar
      ```
  1. Have the cursor at the beginning of file(before F character) Type alt-n, then arrow right, then insert a character. Observe that the character is inserted at the beginning of the line (where the alt-n left you).
  1. Repeat steps using <kbd>alt-o</kbd> instead of <kbd>alt-n</kbd>.

  **Expected**: Both alt-o and alt-n behave the same

  **Issues we've seen:**
  ![bug](https://user-images.githubusercontent.com/5624870/29386722-ebe4c2f6-8291-11e7-88e7-12cc89376cc7.gif)

- [x] Ubuntu: Keystrokes involving <kbd>ctrl</kbd> resolve to the default layout instead of the active layout (https://github.com/atom/atom/issues/13170)

  <details>
  <summary>Repro steps</summary>

  **This should be tested on Linux with gnome**

  1. Install a Linux ditro with gnome, for example Ubuntu
  1. Install French AZERTY and English US QWERTY layouts
  1. Change the default layout to be French AZERTY
  1. Open Atom and switch to English US QWERTY layout
  1. Press ctrl-w

  **Expected**: `core:close` to dispatch (Or the keybinding-resolver to resolve to ctrl-w). Should be resolving to the keyboard layout that is chosen and not the OS default layout.

  **Issues we've seen:** `core:undo` dispatched because it was resolved as ctrl-z because AZERTY has Z where W is on QWERTY.
  </details>

- [x] Other keyboard layouts on new Electron version
  - [x] macOS: ABC - Extended
  - [x] macOS: Slovak
  - [x] macOS: U.S. International
  - [x] macOS: Squirrel (https://github.com/rime/squirrel)
  - [x] macOS: Full-Width Alphanumeric (Google) - Japanese layout  (https://www.google.co.jp/ime/)
  - [x] Windows: U.S.
  - [x] Linux: Pinyin Simplified

### UI

- [x] tree-view drag image (https://github.com/atom/tree-view/pull/1054)
	- [x] macOS
	- [x] Windows 10
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Open a project in Atom with multiple files and a few folders
  1. Drag a file into a folder

  **Expected:**  
  ![image](https://cloud.githubusercontent.com/assets/482957/24454969/3b9e370e-148e-11e7-8757-53171cd6d8e3.png)

  **Issues we've seen:**

  ![image](https://user-images.githubusercontent.com/5624870/30718615-465457ce-9ed5-11e7-96e2-b367fc4d0589.gif)
  </details>

- [x] drag-and-drop indicator (https://github.com/atom/tree-view/pull/1055, https://github.com/atom/tabs/pull/426, https://github.com/atom/tabs/pull/437)
	- [x] macOS
	- [x] Windows 10
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Open Atom with a few folder projects on tree-view (you can do this on terminal in a directory that has existing folder and having a space between each folder. Example: `atom folderA folderB folderC`)
  1. Drag and drop all of the folders to reorder them on Atom

  **Expected:**
  The placeholder indicates where the folder/project is being dropped to.

  ![screen shot 2017-03-29 at 16 31 16](https://cloud.githubusercontent.com/assets/482957/24459995/3e78d498-149d-11e7-8037-dab5a8312822.png)

  **Issues we've seen:**
  No placeholder shows up after drag and dropping

  ![screen shot 2017-03-29 at 16 30 51](https://cloud.githubusercontent.com/assets/482957/24459997/3e9f6b1c-149d-11e7-97c5-683aaa18221a.png)
  </details>

- [x] Large file rendering (https://github.com/atom/atom/issues/16591)
	- [x] macOS
	- [x] Windows 10
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Open a file that has over one million lines
  1. Scroll down to the bottom

  **Expected:** To not regress the number of lines that can be rendered. Rendering to be correct for lines past a certain point. Atom 1.25  can render around 800k to 1 million lines correctly.

  **Issues we've seen:** Increased number of lines rendered but bad rendering past a certain point

  ![bad renderer](https://user-images.githubusercontent.com/1058982/35113772-f2ffceda-fc82-11e7-81b6-6419c42ef3b4.gif)
  </details>

- [x] Loss of subpixel AA when the cursor is at the end of long lines (https://github.com/atom/atom/issues/16889, https://github.com/atom/atom/pull/16595)
	- [x] macOS
	- [x] Windows 10
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Open [text-editor.js](https://github.com/atom/atom/blob/2e19cf4a7c0a4d624b1f9c1072cd292de112cdb0/src/text-editor.js) in Atom
  1. Copy the first 21 lines to a new untitled window
  1. Turn on soft wrapping and change language mode to javascript
  1. Resize the Window so it's soft wrapped
  1. On line 15 type `a` until the you reach the end of the window

  **Expected:** To not lose subpixel AA

  **Issues we've seen:** Loss of subpixel AA. Both when soft wrapping was [enabled](https://github.com/atom/atom/pull/16943) and [disabled](https://github.com/atom/atom/pull/16595).

  ![subpixel aa loss softwrap](https://user-images.githubusercontent.com/1058982/37170628-a3901da0-230b-11e8-991e-c302db473eb3.gif)
  </details>

- [x] Scrolling horizontally shift + scroll wheel (https://github.com/atom/atom/pull/12696#issuecomment-288615912)
	- [x] macOS
	- [x] Windows 10
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Open a file that has long lines
  1. Turn off soft wrapping and resize the window so you have a horizontal scrollbar
  1. Press shift and scroll in both directions using the mouse wheel

  **Expected:** File to scroll horizontally in both directions

  **Issues we've seen:** File does not scroll horizontally at all in any direction
  </details>

- [x] Scrollbars misbehaving on the first file that is opened (https://github.com/atom/atom/pull/12696#issuecomment-251299649)
  <details>
  <summary>Repro steps</summary>

  1. Open Atom
  1. Open some file that has content enough to have a vertical scrollbar
  1. Close Atom
  1. Open Atom and look at the vertical scrollbar

  **Expected:** Scrollbar to be visible

  **Issues we've seen:** Scrollbar is not visible and is flickering when you are editing
  </details>

- [x] Middle clicking on unsaved tab (https://github.com/atom/atom/issues/15197)
	- [x] macOS
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Open Atom and open a new unsaved file.
  1. Make edits to the unsaved file and middle click on the tab.

  **Expected:**
  Clicking save/cancel or the options on the dialog works.

  **Issues we've seen:**
  The UI and the dialog is unable to receive mouse clicks. You can still choose options via Keyboard, but not mouse.
  </details>

- [x] Linux: Atom scrolls even when not focused (https://github.com/atom/atom/issues/15482)
  <details>
  <summary>Repro steps</summary>

  1. Open Atom with a large enough file that has a scrollbar
  1. Open a different application and place it over the Atom window
  1. Scroll with the mouse inside the other window
  1. Focus the Atom window

  **Expected:** Atom window should keep the original scroll position

  **Issues we've seen:** The Atom window scrolls after it is focused
  </details>

### Other

- [x] Deprecation warnings (https://github.com/atom/atom/pull/12696#issuecomment-288871751)
	- [x] macOS
	- [x] Windows 10
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  **This should be tested on macOS, Linux and Windows with community packages installed**

  1. Open Atom
  1. Open the Developer tools console on the master branch and check for deprecation warnings
  1. Open the Developer tools console on the electron upgrade branch and check for deprecation warnings
  1. Compare the two outputs

  **Expected:** No new deprecation warnings

  **Issues we've seen:** New deprecation warnings both from core and community packages
  </details>

- [x] Supported Versions of OS (https://github.com/atom/atom/issues/15297)
	- [x] macOS
	- [x] Windows 10
	- [x] Ubuntu
  <details>
  <summary>Repro steps</summary>

  1. Check the list of supported platforms [Electron Master](https://github.com/electron/electron/blob/master/docs/tutorial/support.md#supported-platforms)
  1. Check what version is listed as minimum for each platform on https://atom.io and https://atom.io/faq

  **Expected:** https://atom.io and https://atom.io/faq to display the same versions as the Electron documentation

  **Issues we've seen:** macOS version support changed and https://atom.io and https://atom.io/faq was not updated
  </details>

- [x] macOS : Native tabs (https://github.com/atom/atom/pull/14711)
  <details>
  <summary>Repro steps</summary>

  1. Open two atom windows on mac
  1. Go to View in menus and check `Show Tab Bar`
  ![screen shot 2017-06-03 at 5 10 32 pm](https://cloud.githubusercontent.com/assets/378023/26751932/aedfc05a-487f-11e7-9643-8047242b3bcb.png)
  1. Do this for both Atom windows.
  1. Drag one atom window to the other via tab bar

  **Expected:**
  Native Tab bar shows up and able to drag different windows to each other.  Also make sure UI isn't messed up.

  **Issues we've seen:**
  Native Tab Bar doesn't work and messes up UI when enabled
  </details>

- [x] Ubuntu with KDE: Menu uses 100% CPU (https://github.com/electron/electron/issues/8455, https://github.com/atom/atom/issues/13885)
  <details>
  <summary>Repro steps</summary>

  1. Install Atom on Ubuntu KDE. [[instructions](https://linuxconfig.org/how-to-install-kde-plasma-desktop-on-ubuntu-18-04-bionic-beaver-linux)]
  1. Enable KDE plasma global menu: Right-click on the desktop -> Add Panel => Application Menu Bar. [[demo](https://youtu.be/qo5fhiXAUbI)]
  1. Start Atom.
  1. Try to use the desktop menu.

  **Expected:**
  Desktop menu items work

  **Issues we've seen:**
  Desktop menu items do nothing when you click on things.
  </details>

- [x] macOS: Slovak QWERTZ (https://github.com/atom/atom-keymap/issues/223)
  <details>
  <summary>Repro steps</summary>

  1. Install Slovak keyboard layout on mac
  1. Open keyboard viewer to see the keys are are suppose to show up and open keyboard resolver on atom.
  1. hit `cmd+'`.

  **Expected:**  Resolves to `cmd+'` like it shows on keyboard viewer on mac.

  **Issues we've seen:** It resolves to `ctrl+§`
  </details>

- [x] macOS: IME jump (https://github.com/atom/atom/issues/15696)
  <details>
  <summary>Repro steps</summary>

  1. Install an IME layout, for example Chinese pinyin
  1. Type one character

  **Expected:**
  Expected the IME to be positioned at under the character you entered
  **Issues we've seen:** The IME window is in the top left corner. When you enter the second character it jumps to be positioned under the text
  </details>

sadick254 committed:
:arrow_up: Upgrade electron@9.3.5

sadick254 committed:
:arrow_up: Bump @atom/fuzzy-native@1.2.0

sadick254 committed:
:arrow_up: Bump superstring@2.4.3

sadick254 committed:
:arrow_up: Bump spell-check@0.76.3

sadick254 committed:
:arrow_up: Bump @atom/nsfw@1.0.27

sadick254 committed:
:arrow_up: Bump @atom/watcher@1.3.5

sadick254 committed:
:arrow_up: Bump scrollbar-style@3.2.2

mfonville commented:
As noted in https://github.com/atom/atom/issues/21422#issuecomment-702402584 and a reminder I had put here: https://github.com/atom-ide-community/atom/pull/75#issuecomment-703079811

If you are upgrading electron to a version of 7 or higher, you can drop the`libgconf-2-4 ` dependency in `resources/linux/debian/control.in`

aminya commented:
For Windows, Superstring error is fixed by: https://github.com/atom/superstring/pull/91

sadick254 committed:
:arrow_up: Bump scrollbar-style@3.2.3

sadick254 committed:
:arrow_up: Bump text-buffer@13.18.3

sadick254 committed:
:arrow_up: Bump superstring@2.4.4

aminya commented:
By the way, probably at some point, you will encounter some errors that say electron-link is not able to bundle the app because of `lazy requires`. You will need to register a new version for the electron-link. My last merged PR is still awaiting registration.

https://github.com/atom/electron-link

which I think it would be better to merge https://github.com/atom/electron-link/pull/31 before registration, as it tests, electron-link for electron 9.

sadick254 commented:
@aminya  Noted.

aminya commented:
> By the way, probably at some point, you will encounter some errors that say electron-link is not able to bundle the app because of `lazy requires`. You will need to register a new version for the electron-link. My last merged PR is still awaiting registration.
> 
> https://github.com/atom/electron-link
> 
> which I think it would be better to merge [atom/electron-link#31](https://github.com/atom/electron-link/pull/31) before registration, as it tests electron-link for electron 9.

I believe the current error in the CI is the error that I am talking about. 

> Assigning a deferred module to a variable that was not declared in this scope is not supported!

sadick254 committed:
:arrow_up: Bump electron-link@0.5.0

sadick254 commented:
@aminya Am afraid the error on the CI is not the one you are talking about. I have however published a new release for `electron-link` and included it on this PR. The build is throwing an error while dumping symbols.

mfonville commented:
I notice btw that in the bootstrap phase (at least on Linux) Python is on 2.7; shouldn't it be already be Python3 with these higher electron/apm/nmp versions?

aminya commented:
On Windows, less-node is throwing errors. 

> ##[warning] The reference to the module is replaced with the lazy function, but it is assigned to "module" or "exports". In some cases the bundle might not work, which you should fix manually: 
 D:\a\1\s\out\app\node_modules\less\lib\less-node\fs.js 


Probably my Less 3 PR will fix that:
https://github.com/atom/less-cache/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc

aminya commented:
Minidump error on Linux and MacOS will be probably fixed by https://github.com/atom/atom/pull/21792

mfonville commented:
Electron 9.4.0 is now also available with some (security) fixes

sadick254 committed:
:arrow_up: Bump electron-link@0.6.0

aminya commented:
The current failure in Windows CI is because of the "debug" module in spellcheck package. I have opened an issue for it. 

https://github.com/atom/spell-check/issues/356

sadick254 committed:
:arrow_up: Bump spell-check@0.77.0

aminya commented:
Windows build in PR CI finished. 🎉 However, one of the tests time outs.

aminya commented:
`fswin` issue in Linux/MacOS is because of `winattr` which is used in `text-buffer`. I made a PR which probably fixes it.
https://github.com/atom/text-buffer/pull/336

aminya commented:
Running the build localy on Windwos fails because of the old superstring which is becaue of the old whats-my-line in github package. 

We need this PR to fix that:
https://github.com/atom/atom/pull/21782

aminya committed:
fix spell-check version in package.json

aminya commented:
Windows build in PR CI finished. However, one of the tests time outs, which is launching the Atom window! 

I built Atom locally after applying the changes of [this comment](https://github.com/atom/atom/pull/21777#issuecomment-746936053).
However, Atom does not open:

![browser](https://user-images.githubusercontent.com/16418197/102403191-7e8c1800-3fab-11eb-8f1c-addda34d5be8.gif)


-----
Also, Production CI fails to sign the executable due to a connection issue. Is the link still valid?

aminya commented:
Downgrading to electron 7.3.3 gives a working Atom. However, I get these warnings when I launch Atom in dev mode:
```js
(node:32076) Electron: Loading non context-aware native modules in the renderer process is deprecated and will stop working at some point in the future, please see https://github.com/electron/electron/issues/18397 for more information
```
and when I closed Atom:
```js
Attempting to call a function in a renderer window that has been closed or released.
Function provided here: Object.<anonymous> (C:\Users\aminy\Documents\GitHub\JavaScript\@atom-ide-community\atom\node_modules\github\lib\worker.js:60:22
Remote event names: crashed, destroyed
```

aminya commented:
`app.allowRendererProcessReuse` shuold be set to false in Electron 9, so we can keep using non-contet aware native modules.

Trying Electron 8.5.5 gives better warning messages about these native modules that should be updated. Pretty much all non `N-api` packages. If using NAN we should apply these changes like this: https://github.com/atom/node-keytar/pull/182

This should be done before Electron 12 based on https://github.com/electron/electron/issues/18397 

```
❯ (electron) The default value of app.allowRendererProcessReuse is deprecated, it is currently "false".  It will change to be "true" in Electron 9.  For more information please check https://github.com/electron/electron/issues/18397
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\scrollbar-style\build\Release\scrollbar-style-observer.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\text-buffer\node_modules\superstring\build\Release\superstring.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\pathwatcher\build\Release\pathwatcher.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\git-utils\build\Release\git.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\tree-sitter-json\build\Release\tree_sitter_json_binding.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\oniguruma\build\Release\onig_scanner.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\Users\aminy\Documents\GitHub\JavaScript\@atom\spell-check\node_modules\spellchecker\build\Release\spellchecker.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\Users\aminy\.atom\packages\Hydrogen\node_modules\zeromq\build\Release\zmq.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\tree-sitter-c\build\Release\tree_sitter_c_binding.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\tree-sitter-cpp\build\Release\tree_sitter_cpp_binding.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\tree-sitter-css\build\Release\tree_sitter_css_binding.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\tree-sitter-go\build\Release\tree_sitter_go_binding.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\tree-sitter-embedded-template\build\Release\tree_sitter_embedded_template_binding.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\tree-sitter-html\build\Release\tree_sitter_html_binding.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\tree-sitter-java-dev\build\Release\tree_sitter_java_binding.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\tree-sitter-javascript\build\Release\tree_sitter_javascript_binding.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\tree-sitter-jsdoc\build\Release\tree_sitter_jsdoc_binding.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\tree-sitter-regex\build\Release\tree_sitter_regex_binding.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\tree-sitter-python\build\Release\tree_sitter_python_binding.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\tree-sitter-ruby\build\Release\tree_sitter_ruby_binding.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\tree-sitter-rust\build\Release\tree_sitter_rust_binding.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\tree-sitter-bash\build\Release\tree_sitter_bash_binding.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\tree-sitter-typescript\build\Release\tree_sitter_tsx_binding.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\tree-sitter-typescript\build\Release\tree_sitter_typescript_binding.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:31924) Electron: Loading non-context-aware native module in renderer: '\\?\C:\atom\out\Atom Dev x64\resources\app.asar.unpacked\node_modules\@atom\nsfw\build\Release\nsfw.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
```

DeeDeeG commented:
`electron.asar` was removed in Electron 7. See: https://github.com/electron/electron/pull/18577

[`src/module-cache.js`](﻿﻿﻿﻿https://github.com/atom/atom/blob/v1.53.0/src/module-cache.js) refers to `electron.asar` and should be updated. Perhaps some packages need to be updated for this change as well.

aminya committed:
set app.allowRendererProcessReuse to false

so we can keep using non-contet aware

TODO this should be set to true before Electron 12

https://github.com/electron/electron/issues/18397

https://github.com/atom/atom/pull/21777#issuecomment-747066738

aminya commented:
Here we go. The first working Atom Electron 9 on Windows after merging the #21805.

![electron9](https://user-images.githubusercontent.com/16418197/102417218-e26d0b80-3fc0-11eb-8d4d-a33d99458ac7.gif)

DeeDeeG commented:
>/Applications/Atom Dev.app/Contents/Resources/app.asar/src/text-editor-element.js:357
>
>[Deprecation] document.registerElement is deprecated and will be removed in M80, around February 2020. Please use window.customElements.define instead. See https://www.chromestatus.com/features/4642138092470272 and https://developers.google.com/web/updates/2019/07/web-components-time-to-upgrade for more details.

This deprecation affects Chrome 80 and newer. (Which should mean Electron 9 and newer? Unless Electron continues to support `document.registerElement`, for example by patching Chromium?)

The following files might need to be updated:

```console
% git grep --name-only document.registerElement
spec/panel-container-element-spec.js
spec/text-editor-component-spec.js
spec/text-editor-element-spec.js
src/pane-axis-element.coffee
src/pane-container-element.js
src/pane-element.js
src/pane-resize-handle-element.coffee
src/panel-container-element.js
src/styles-element.coffee
src/text-editor-element.js
src/workspace-element.js
```

mfonville commented:
Great to see good results @aminya 

I thought, maybe also an idea for the devs to directly clean-up some old Electron-TODOs that were marked for when Electron would be upgraded (most are left-overs from version 3):
https://github.com/atom/atom/blob/0226ae9ffa06fae63bafaf528663e4c3eca8dc0f/src/native-compile-cache.js#L40
https://github.com/atom/atom/blob/50a74a5e31eb767a302da12d671d7450b25cf224/spec/integration/helpers/start-atom.js#L62
https://github.com/atom/atom/blob/5ed857289f86fd5037dc4dad8a394619cb30a03a/src/main-process/atom-application.js#L110
https://github.com/atom/atom/blob/5302cf92b3872bdad8957c6d51f2d29137553466/src/task.coffee#L87

Please also don't forget my earlier comment about dropping the libgconf-2-4 dependency :-) (and probably the RPM dependencies need a refresh, but I am not acquainted with it)

aminya commented:
> Please also don't forget my earlier comment about dropping the libgconf-2-4 dependency :-) (and probably the RPM dependencies need a refresh, but I am not acquainted with it)

@mfonville Maybe open a PR dropping the dependency you are talking about and target this branch in your PR.

aminya committed:
Rename TODO to TodoElectronIssue so it can be found later

aminya commented:
> `electron.asar` was removed in Electron 7. See: [electron/electron#18577](https://github.com/electron/electron/pull/18577)
> 
> `src/module-cache.js` refers to `electron.asar` and should be updated. Perhaps some packages need to be updated for this change as well.

Fixed in https://github.com/atom/atom/pull/21805/commits/affedde09325f7f67763381705da080392f1d301

aminya commented:
> > /Applications/Atom Dev.app/Contents/Resources/app.asar/src/text-editor-element.js:357
> > [Deprecation] document.registerElement is deprecated and will be removed in M80, around February 2020. Please use window.customElements.define instead. See https://www.chromestatus.com/features/4642138092470272 and https://developers.google.com/web/updates/2019/07/web-components-time-to-upgrade for more details.
> 
> This deprecation affects Chrome 80 and newer. (Which should mean Electron 9 and newer? Unless Electron continues to support `document.registerElement`, for example by patching Chromium?)
> 

It is still working in Electron 9. 
![image](https://user-images.githubusercontent.com/16418197/102422329-23b6e880-3fcc-11eb-92cf-5443de059f9f.png)

aminya committed:
Fix remote deprecation

remote is deprecated

https://www.electronjs.org/docs/breaking-changes#deprecated-remote-module

Use require("electron").remote instead of require("remote")

aminya committed:
Fix shell deprecation

shell is deprecated

Use require('electron').shell instead

aminya committed:
Remove electron.asar caching

electron.asar was removed in Electron 7: 

electron/electron#18577

mfonville commented:
@aminya I now filed the PR for the dependency update

sadick254 committed:
Ignore fswin.node on non-windows platform

fswin.node is only used on windows, ignore the error on other platforms

sadick254 committed:
:arrow-up: Bump text-buffer@13.18.5

aminya committed:
Add toBeNear jasmine matcher

aminya commented:
I fixed some of the text-component spec on Windows using #21811

aminya committed:
Use toBeNear in the text-editor-component-spec

aminya committed:
Use toBeNear in more places in text-editor-component spec

These are required for passing local tests

aminya committed:
Add toHaveNearPixels for comparing pixel strings

mfonville committed:
Drop libgconf dependency for electron7+ (#21806)

* Drop libgconf dependency for electron7+

* Also remove libgconf from preparation template

DeeDeeG commented:
I made small fix in the `autocomplete-plus` package, as follow-up to #21805. Here is the PR: https://github.com/atom/autocomplete-plus/pull/1096

sadick254 committed:
:arrow_up: Bump autocomplete-plus@2.4.4

aminya committed:
Use parseFloat in toHaveNearPixels

sadick254 committed:
Use nativeImage for linux icon

mfonville commented:
I made a PR that should deal with the 2nd and 3rd todoElectron mentioned in https://github.com/atom/atom/pull/21777#issuecomment-747102989

For the 1st and 4th I am not acquainted enough daring to make the necessary changes (also can't do any local testing here). But I assume the first one should be as trivial as in the example described in the documentation here: https://nodejs.org/api/vm.html#vm_script_createcacheddata

and for the 4th probably just removing the parameter is all that is necessary.
Maybe @aminya or @DeeDeeG have time to take a look? :-)

sadick254 committed:
Fix linting

DeeDeeG commented:
There has been an issue with the Production and Nightly pipelines. They can't download the code signing certificates (404 error), so `script/build` fails. I posted an issue about it: https://github.com/atom/atom/issues/21817

_(It's arguably off-topic for this PR, but without fixing it, making an Electron 9 release would be blocked. Or any release, for that matter!)_

sadick254 commented:
@DeeDeeG We are aware of this and trying to fixit. But thanks for bringing it up. We will surely fix it before merging this to master.

sadick254 committed:
Refactor text-editor-element-spec

sadick254 committed:
Only pass neccessary env variables

sadick254 committed:
Use toBeNear matcher

sadick254 committed:
Update expected values

sadick254 committed:
:arrow_up: Bump settings-view

aminya commented:
@sadick254 Whenever you bump an atom package, you also need to bump in the packageDependencies section. So, it needs bumping in two places.
https://github.com/atom/atom/blob/cdbf8ffe83e7edf3d2d6b50598f0c7ef9d088a6b/package.json#L222

DeeDeeG commented:
By the way: Does anyone know exactly what the `packageDependencies` are used for? (I think `apm` reads them and makes sure they're included... But when we have it in `dependencies` _and_ `packageDependencies`, I wonder if `apm` just copies the version from the `node_modules` folder...) I tried doing a code search for `packageDependencies` and it's referenced in several places.

aminya commented:
For the failing GitHub tests please see this comment:
 https://github.com/atom/atom/pull/21782#issuecomment-753312833

mfonville committed:
Electron 9.4.1 (#21815)

* Upgrade to 9.4.1

mfonville commented:
https://github.com/electron/electron/releases/tag/v9.4.2
9.4.2 arrived

asturur commented:
If you need some help i can help with regression testing Macos or windows 10

DeeDeeG commented:
Hi again @asturur. Here's what I think is left to do for Electron 9:

- Make the automated tests pass
- Manually test a bunch of things that have broken in past Electron upgrades
- More manually testing stuff to make sure nothing else is broken

At the moment, the automated tests are not all passing. So I think that would be the most-focused-on thing. Once all the automated testing is clear, then it is easier to work on other problems, because you can change things and expect CI to pass again even after you fix the other things.

---

For automated tests: I think it's just down to one failing test on Windows. The latest `github` package 0.36.7 should fix the automated test failure on macOS.

asturur commented:
Well i did not want to put any pressure. I wanted to say that if you need some manual regression test on mac or windows i m available

the-j0k3r commented:
I thought electron bumps were done by 1 major version at a time until we reach the latest electron version?

DeeDeeG commented:
@the-j0k3r I think the developers realized that Electron has a more rapid pace of releases now. Several releases in a row have been based on Node 12. And there is a new person working on this. Anyhow, trying something new.

9.4.1 targeted the branch for this PR, so it was merged into this, not into `master`.

the-j0k3r commented:
> For what it's worth, I've tried building this branch against Electron 7 to see if it works any better, and so far it doesn't seem to work any better than Electron 9. I think going to Electron 9 is a reasonable move if it works and bugs can be squashed.

Magic words identified. ` if it works and bugs can be squashed.`

I think its a good idea to try and test, but I wonder about those bugs you mentioned where plugins are concerned and not just Atom side.

Anyway, good job, Im a fan of keeping dependencies up-to-date wherever and electron is the most crucial for Atom.

Watching this thread, sitting at the edge of my chair. =)

mfonville commented:
Indeed electron being based for several releases on node12 should probably alleviate the pains for upgrading the dependency, with less changes that have to be accounted for.
But also, there was a huge Electron-backlog. The dependency was severely outdated for a long while (including the fact that there were no security updates anymore).
On 2nd of March Electron 12.x will go stable, which I think implies even Electron 9.x support will be dropped at that point (they only support the latest 3 major releases). So trying to catch up and moving to Electron 11.x, which is the latest node12, as swiftly as possible is the right thing to do imho.

sadick254 commented:
Has anyone managed to get a working windows build on this branch?

sadick254 committed:
enableRemoteModule

sadick254 committed:
Remove failing tests on windows.

Due to the upgrade of electron to version 9.4.1, this test has been
failing on windows platform only. Looking at the failing test, it is
clear that we are already testing the softWrap functionality. There is
no need to retest this functionality.

aminya commented:
> Has anyone managed to get a working windows build on this branch?

Yes! The first working build was actually on Windows:
https://github.com/atom/atom/pull/21777#issuecomment-747094244

DeeDeeG commented:
I just built a pretty recent version of this branch on Windows, and Atom started in the background, but no visible windows showed. I might try again on another computer to confirm... But yeah, there may be an issue.

aminya commented:
It works for me. Just download the latest artifact from the CI.

DeeDeeG commented:
Maybe it's because `autocomplete-plus` wasn't updated in the `packageDependencies` section of the main `package.json`. 

I used `script/build` without the `--ci` flag. Apparently the old version `2.42.3` from `packageDependencies` overwrote the newer version `2.42.4` in the regular `dependencies`. So that should be updated.

sadick254 committed:
:arrow_up: Bump autocomplete-plus

sadick254 commented:
@DeeDeeG Looks like I forgot about that while bumping `autocomplete-plus`. `autocomplete-plus` has now been updated.

DeeDeeG commented:
> opening a project doesn't work

`dialog.showOpenDialog()` and a few other parts of the Electron API no longer officially accept callbacks, and any callbacks passed to them aren't run. That's what broke opening projects, apprently. These API calls must be done with Promises now, not callbacks. (Or I think `async`/`await` can be a valid approach too, as that works with Promises? Someone good with JS can confirm?)

I have a fix for this particular "can't open a project" issue, (https://github.com/atom/atom/pull/21926, reviews appreciated) but there are a number of other Electron API calls where passing a callback is deprecated. See: https://www.electronjs.org/docs/api/modernization/promisification

___

To track down these changed/deprecated/removed APIs, I've been copy-pasting the name of every entry on the promisification docs page, and doing a code search `git grep [-i for case-insensitive search if you want] [search term here, may be a good idea to put in single quotes]`

For example: `git grep -i showOpenDialog`

asturur commented:
i can try to fix the. callback vs promise, all of them with an extensive search, if you do not need it in a rush. I'm ok-skilled with JS

DeeDeeG commented:
@asturur I noticed the save dialog also is affected, if you want to give fixing that a shot. `git grep saveItem` (also `saveItemAs`) and `git grep showSaveDialog` will show most of the uses/the functions that call into the save dialog function.

That one is a bigger web of several functions, some of which are public Atom API functions that we might want to preserve the existing behavior for them if possible, but hopefully someone with more-advanced JS than me can handle it.

sadick254 commented:
@DeeDeeG We still have some timing out tests on the github package. I know you have done some extensive research on this recently. The timeout could be due to the way github package tests are executed. Any pointers on this?

@asturur I am happy to review your open PR's on the callback to promise transforms.

DeeDeeG commented:
@sadick254 I can try a CI run with `github` tests in their own job to see if the error messages are more specific. Like https://github.com/atom/atom/pull/21899. And if the messages are useful I'll report them back here.

asturur commented:
i ll give it a shot on the weekend

aminya commented:
@sadick254 Should we cherry-pick some of the changes in this PR and merge them to the master branch? This helps to reduce the friction of this pull request and reduce the number of things that should be checked and tested. It also makes it easier to pinpoint the issues.

This PR may take some time until it gets merged, but some of these commits can be merged safely to the master branch.

sadick254 commented:
@aminya  The commits `95e2c01..e37eee2` are touching on tests and matchers. Those commits have no direct impact on atom users as far as I can tell. I prefer having all the commits related to updating electron on this branch as opposed to merging some on `master`. That way the context of the commits are not lost.

mfonville commented:
A suggestion for the promise changes:
they could be tested for the current master branch, based on electron 6.1.12. Tthere already many functions converted and if you enable the `process.enablePromiseAPI` flag during testing you get deprecation notices: https://github.com/electron/electron/blob/v6.1.12/docs/api/modernization/promisification.md

That way can you more easily track any bugs/regressions when you are changing these callbacks to promises.

DeeDeeG commented:
Here are the error messages for the `github` package tests:

<details><summary>Error messages for the github package tests (click to expand):</summary>

```
  2441 passing (15m)
  1 pending
  4 failing

  1) GithubDotcomMarkdown
       opening issueish links
         opens item in pane and activates accordingly:
     Error: Trying to stub property 'open' of null
      at throwOnFalsyObject (/Users/runner/work/1/s/node_modules/github/node_modules/sinon/lib/sinon/throw-on-falsy-object.js:7:15)
      at Function.stub (/Users/runner/work/1/s/node_modules/github/node_modules/sinon/lib/sinon/stub.js:70:24)
      at Sandbox.stub (/Users/runner/work/1/s/node_modules/github/node_modules/sinon/lib/sinon/sandbox.js:331:37)
      at Context.<anonymous> (/Users/runner/work/1/s/node_modules/github/github-dotcom-markdown.test.js:143:13)
      at processImmediate (internal/timers.js:439:21)

  2) GithubDotcomMarkdown
       opening issueish links
         records event for opening issueish in pane item:
     Error: Trying to stub property 'open' of null
      at throwOnFalsyObject (/Users/runner/work/1/s/node_modules/github/node_modules/sinon/lib/sinon/throw-on-falsy-object.js:7:15)
      at Function.stub (/Users/runner/work/1/s/node_modules/github/node_modules/sinon/lib/sinon/stub.js:70:24)
      at Sandbox.stub (/Users/runner/work/1/s/node_modules/github/node_modules/sinon/lib/sinon/sandbox.js:331:37)
      at Context.<anonymous> (/Users/runner/work/1/s/node_modules/github/github-dotcom-markdown.test.js:167:13)
      at processImmediate (internal/timers.js:439:21)

  3) GithubDotcomMarkdown
       opening issueish links
         does not record event if opening issueish in pane item fails:
     Error: Trying to stub property 'open' of null
      at throwOnFalsyObject (/Users/runner/work/1/s/node_modules/github/node_modules/sinon/lib/sinon/throw-on-falsy-object.js:7:15)
      at Function.stub (/Users/runner/work/1/s/node_modules/github/node_modules/sinon/lib/sinon/stub.js:70:24)
      at Sandbox.stub (/Users/runner/work/1/s/node_modules/github/node_modules/sinon/lib/sinon/sandbox.js:331:37)
      at Context.<anonymous> (/Users/runner/work/1/s/node_modules/github/github-dotcom-markdown.test.js:179:13)
      at processImmediate (internal/timers.js:439:21)

  4) WorkerManager
       when the manager process is destroyed
         destroys all the renderer processes that were created:
     Error: Timeout of 60000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/Users/runner/work/1/s/node_modules/github/test/worker-manager.test.js)
```

</details>

asturur commented:
I cloned, i swapped the first showOpenDialog, i ll give a run to cover the promisified functions extensively this saturday.

asturur commented:
The issue is that i have to take for granted the documentation of 5.x because in version 6, when they added the promise, the callback became optional and lost the description of the arguments.

mfonville commented:
Electron 9.4.3 is in with some security fixes

DeeDeeG commented:
When building locally, I can't get this branch to launch any windows. On Windows and on macOS. 🤷

asturur commented:
neither on macos. Crash immediately. I indeed tested my fixes on master.

DeeDeeG commented:
To narrow down the problem: I have a branch with a few less package upgrades and with Electron 7. I can try to figure out if it was a package upgrade that broke things, or if Electron 7 fixes it, or if even downgrading back to Electron 9.3.5 fixes it, and so on...

And I should check if the CI artifacts work, because that would be another lead to look into.

DeeDeeG commented:
Thanks for the encouragement, @the-j0k3r

sadick254 committed:
Update showOpenDialog to promise.

sadick254 commented:
@DeeDeeG locally building Atom on my local machine works just fine on macOS. Are you getting any errors on macOS?

DeeDeeG commented:
@sadick254 I can try it again tomorrow. I was having trouble before (a few commits back) when building locally.

DeeDeeG commented:
Okay, it looks like the build from this branch doesn't like to open if there are no projects in the `tree-view` area (the "Project" sidebar on the left). Or stored in the application state (no `projectRoots` entries in `[home_dir]/.atom/storage/application.json`).

If you launch it with a folder to add on the CLI, with `-a` or `--add` (like this: `/path/to/atom --add path/to/some_folder/here`) then it launches. Or if you open a working copy of Atom and add a project, then you can successfully launch the copy of Atom built from this branch. Or if you launch this build of Atom by dragging a folder onto it, it will work.

Anything that adds a proper `projectRoots` entry in `.atom/storage/application.json` makes this build of Atom able to open windows again.

Tested and acts the same on Linux, macOS and Windows.

sadick254 committed:
Update resolveProxy to promise

asturur commented:
Gave a check to all remaining promisification items and seems that everything is covered.

Is there anything that can be picked up to help? a particular test or a particular bug

DeeDeeG commented:
@asturur Some of the `github` package tests are failing for unknown reasons. But there are some outdated Electron API uses directly in the `github` package. So maybe that is why the tests are failing?

Efforts to address that are being discussed here: https://github.com/atom/github/issues/2624

asturur commented:
related PR atom/github#2626, i hope is useful.
If you want to give me a specific task, try, maybe i have time to do it in weekends or some evenings.

asturur committed:
⬆️ Bump github@0.36.8 (#21975)

sadick254 committed:
Check before parsing

DeeDeeG commented:
With the latest commit https://github.com/atom/atom/commit/78c1beda471d81429f03b66645cd8cd40b9eb728, I can open Atom when there is a blank `projectRoots` in `.atom/storage/application.json`. :tada:

I do see a couple of errors in any window launched with blank `projectRoots`, though:

<details>

```
Failed to load a language-hyperlink package grammar

Cannot read property 'id' of undefined in /Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app.asar/node_modules/language-hyperlink/grammars/hyperlink.json

TypeError: Cannot read property 'id' of undefined
    at Function.startNextJob (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:14:915053)
    at SpellCheckTask.start (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:14:914555)
    at SpellCheckView.updateMisspellings (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:958275)
    at /Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:957703
    at Function.simpleDispatch (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:1182710)
    at Emitter.emit (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:1184151)
    at /Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:1:802120
    at Function.simpleDispatch (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:1182710)
    at Emitter.emit (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:1184151)
    at Text[user]LanguageMode.markTokenizationComplete (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:557198)
    at Text[user]LanguageMode.retokenizeLines (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:556151)
    at Text[user]LanguageMode.updateForInjection (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:555941)
    at /Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:360694
    at Map.forEach (<anonymous>)
    at GrammarRegistry.grammarAddedOrUpdated (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:360299)
    at Function.simpleDispatch (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:1182710)
    at Emitter.emit (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:1184151)
    at GrammarRegistry.addGrammar (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:14:120788)
    at Grammar.activate (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:14:127970)
    at /Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:3375172
    at /Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:362234
    at u (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:583651)
    at /Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:584904
    at electron/js2c/asar.js:533:9
    at FSReqCallback.wrapper [as oncomplete] (fs.js:470:5)
  at /Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app.asar/node_modules/language-hyperlink/grammars/hyperlink.json:1:1
```

```
Failed to load a language-todo package grammar

Cannot read property 'id' of undefined in /Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app.asar/node_modules/language-todo/grammars/todo.json

TypeError: Cannot read property 'id' of undefined
    at Function.startNextJob (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:14:915053)
    at SpellCheckTask.start (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:14:914555)
    at SpellCheckView.updateMisspellings (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:958275)
    at /Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:957703
    at Function.simpleDispatch (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:1182710)
    at Emitter.emit (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:1184151)
    at /Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:1:802120
    at Function.simpleDispatch (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:1182710)
    at Emitter.emit (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:1184151)
    at Text[user]LanguageMode.markTokenizationComplete (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:557198)
    at Text[user]LanguageMode.retokenizeLines (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:556151)
    at Text[user]LanguageMode.updateForInjection (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:555941)
    at /Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:360694
    at Map.forEach (<anonymous>)
    at GrammarRegistry.grammarAddedOrUpdated (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:360299)
    at Function.simpleDispatch (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:1182710)
    at Emitter.emit (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:1184151)
    at GrammarRegistry.addGrammar (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:14:120788)
    at Grammar.activate (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:14:127970)
    at /Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:3375172
    at /Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:362234
    at u (/Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:583651)
    at /Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app/static/<embedded>:11:584904
    at electron/js2c/asar.js:533:9
    at FSReqCallback.wrapper [as oncomplete] (fs.js:470:5)
  at /Users/[user]/Downloads/9-projectRoots-fix/Atom Dev.app/Contents/Resources/app.asar/node_modules/language-todo/grammars/todo.json:1:1
```

</details>

These are in the `language-hyperlink` and `language-todo` packages. Any window where there was something in `projectRoots` does not show those errors.

sadick254 committed:
:arrow_up: Bump spell-check@0.77.1

sadick254 committed:
:arrow_up: Bump github@0.38.9

sadick254 committed:
Set --auto option

sadick254 committed:
Clang 9 is already in the image.

It's on the PATH and is the version that will be used without needing to do update-alternatives.

mfonville commented:
Electron went to 9.4.4:
https://github.com/atom/atom/pull/22003

mfonville committed:
Electron 9.4.4 (#22003)

aminya commented:
Thanks for the work! 🎉

patricecoulon commented:
J'ai beau me désenregistrer de votre site, ça continue.
Et bien NIQUEZ VOTRE MOHAMMED EN POSITION PRIERE DE DROMADAIRE !
Sourate 9 verset 5 et 30 : BAISE TA MERE SALE MUSULMAN !

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐
Le vendredi 5 mars 2021 18:30, Amin Yahyaabadi <notifications@github.com> a écrit :

> Thanks for the work! 🎉
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, [view it on GitHub](https://github.com/atom/atom/pull/21777#issuecomment-791566712), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/ARUUE7UGYH77YEW7NH5AFDDTCEIJVANCNFSM4UJKSVAA).

DeeDeeG commented:
Wow! Electron 9 on `master` branch!! 🎉 🎉

the-j0k3r commented:
@UziTech interesting to observe effect on terminus/x-terminal

aminya commented:
> interesting to observe effect on terminus/x-terminal as well as all built in Atom addons though.
> 

If we integrate #21927, there will be no issues

the-j0k3r commented:
> If we integrate #21927, there will be no issues

indeed but only if that PR is merged into master before master code becomes a stable release. so we hope it is indeed merged asap.
